opensir/models/post_regression.py

```diff
@@ -31,190 +31,163 @@ def percentile_to_ci(alpha, p_bt):
-        function of time. A common use of this list is to plot the
-        mean squared error against time, to identify time periods
-        where the model produces the best and worst fit to the data.
-    """
-
-    p0 = model.p
-    w0 = model.w0
-
-    if options is None:
-        options = {"lags": 1, "min_sample": 3}
-
-    lags = options["lags"]
-
-    # Consider at least the three first datapoints
-    p_list = []
-    mse_list = []  # List of mean squared errors of the prediction for the time t+1
-    for i in range(options["min_sample"] - 1, len(n_i_obs) - lags):
-        # Fit model to a subset of the time-series data
-        model.fit(t_obs[0:i], n_i_obs[0:i], population)
-        # Store the rolling parameters
-        p_list.append(model.p)
-        # Predict for the i+1 period
-        model.solve(t_obs[i + lags], numpoints=int(t_obs[i + lags]) + 1)
-        pred = model.fetch()[:, 2]
-        # Calculate mean squared errors
-        mse = np.sqrt((pred[i - 1 + lags] - n_i_obs[i - 1 + lags]) ** 2)
-        mse_list.append(mse)
-
-    p_list = np.array(p_list)
-    mse_list = np.array(mse_list)
-    mse_avg = np.mean(mse_list)
-
-    model.p = p0
-    model.w = w0
-
-    return mse_avg, mse_list, p_list
+        Args:
+
+            t_obs (list or np.array): List of days where the number of infected
+                where measured
+
+            n_i_obs (list or np.parray): List with measurements of number of
+                infected people in a population. Must be consistent with t_obs.
+
+            population (int): population size
+
+            options (dict): Time bootstrapping options
+                - lags (int): Defines the number of days that will be
+                forecasted to calculate the mean squared error. For
+                example, for the prediction Xp(t) and the real value
+                X(t), the mean squared error will be calculated as mse =
+                `1/n_boots |Xp(t+lags)-X(t+lags)|`. This provides an
+                estimate of the mean deviation of the predictions after
+                "lags" days.  (default = 1)
+                - min_sample (int): Number of days that will be used in the train
+                set to make the first prediction. (default = 3)
+
+
+        Returns:
+            float: Simple average of the mean squared error between the model
+                prediction for "lags" days and the real observed value.
+
+            numpy.array: List of the mean squared errors using (i) points to
+                predict the X(i+lags) value, with i an iterator that goes from
+                n_samples+1 to the end of t_obs index.
+
+            numpy.array: List of the parameters sampled on the bootstrapping as
+                a function of time. A common use of this list is to plot the
+                mean squared error against time, to identify time periods where
+                the model produces the best and worst fit to the data.
```

Comment:
Here I think you forgot to call the name of the returns. Please rewrite mse_avg, mse_list and p_list before the type.

Reply:
the returns shouldn't have names, because these variables are not exposed to the user (see [here](https://pythonhosted.org/an_example_pypi_project/sphinx.html)).

Strictly speaking the function only returns one variable (a list or tuple in this case). I will update that.

Reply:
Why? it should return three values to the user, the ci, the list of the MSE and the list of parameters. All three returns are of interest when applying block_cv. Do you mean that all that is packed?

Reply:
Yes, all of that is packed. It's just one "tuple" of 3 elements.

What I'm doing now is adding the tuple, and document each field separately. But for effects of Sphinx, the output is "one tuple"

Reply:
I just pushed a fixup with the last proposal. It reduces the documentation to "one" argument, but I expanded the list on purpose to reflect what each term means

Reply:
Cool! now everything is clearer, thank you!